lib/install/deps.js

```diff
@@ -600,6 +600,7 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
     if (typeof child.package.bin !== 'object') return false
+    if (!pkg.bin || !child.package.bin) return false
```

Comment:
@iarna Makes sense. Will definitely fix that. I was actually thinking the same, but I thought there are other reasons to check against the type. 👍